trustfall_core/src/schema/mod.rs

```diff
@@ -315,9 +315,10 @@ fn check_root_query_type_invariants(
                 field_type.to_string(),
             ));
         } else if !vertex_types.contains_key(base_named_type) {
-            // Somehow the base named type is neither a vertex nor a scalar,
-            // and this field is neither an edge nor a property.
-            unreachable!()
+            errors.push(InvalidSchemaError::UnknownPropertyOrEdgeType(
+                field_defn.node.name.to_string(),
+                field_type.to_string(),
+            ))
```

Comment:
I'd do two things:
- I'd delete this entire `else if` block and replace it with a comment that says that the `vertex_types.contains_key(base_named_type)` invariant is enforced by the `check_type_and_property_and_edge_invariants` function, and is tested by the test by such and such name.
- Then make sure there are tests for both individual cases (on root / not on root), as well as one test that has this error both on the root and not on the root so that we have that behavior pinned down as well. In that last one, I think it's fine if only the root error is reported — it's more about knowing precisely what happens and being able to notice if it changes.